File: PythonProgramming/DifferentStrings.py
class DifferentStrings:
    def __init__(self, string1, string2):
        self.s1 = string1
        self.s2 = string2
    
    def minimize(self):
        self.count = 0
        if (len(self.s1) != len(self.s2)):
            self.longestStringLength = max(len(self.s1), len(self.s2))
            if(len(self.s1) < len(self.s2)):
                self.numNull = self.longestStringLength - len(self.s1)
                self.s1 = self.s2[0:self.numNull] + self.s1
            else:
                self.numNull = self.longestStringLength - len(self.s2)
                self.s2 = self.s2 + self.s1[len(self.s1) - self.numNull : len(self.s1)]
        
        for i in range(0, len(self.s1)):
            if(self.s1[i] != self.s2[i]):
                self.count = self.count + 1
        
        return self.count

if __name__ == '__main__':
    value = DifferentStrings("mother", "hoodfather")
    print value.minimize()
    

    
    

File: JavaProgramming/DifferentStrings.java
package com.tutorialspoint;
import java.io.*;
import java.util.Arrays;
import java.lang.*;
public class DifferentStrings{
	public int minimize(String string1, String string2){
		String s1 = string1;
		String s2 = string2;
		char[] S1 = s1.toCharArray();
		char [] S2 = s2.toCharArray();
		
		int longestStringLength, numNull = 0;
		int count = 0;
		
		StringBuilder sb = new StringBuilder(64);
		
		if(S1.length != S2.length){
			longestStringLength = Math.max(S1.length, S2.length);
			if(S1.length < S2.length){
				numNull = longestStringLength - S1.length;
				char [] temp = Arrays.copyOfRange(S2, 0, numNull);
				sb.append(temp);
				sb.append(S1);
				S1 = sb.toString().toCharArray();
			}
			else{
				numNull = longestStringLength - S2.length;
				char [] temp = Arrays.copyOfRange(S1, S1.length - numNull, S1.length);
				sb.append(S2);
				sb.append(temp);
				S2 = sb.toString().toCharArray();
			}
		}
		System.out.println(S1);
		System.out.println(S2);
		
		for (int i = 0; i < S1.length; i ++){
			if(S1[i] != S2[i])
				count++;
		}
		
		return count;
	}
	
	public static void main(String args[]){
		DifferentStrings ds = new DifferentStrings();
		String one = "motherhood";
		String two = "father";
		int result = ds.minimize(one, two);
		System.out.println("Difference between the two strings: " + result);
	}
}